goscraper.go

```diff
@@ -19,6 +19,63 @@ var (
 	fragmentRegexp         = regexp.MustCompile("#!(.*)")
 )
 
+type scrapeSettings struct {
+	userAgent         string
+	maxDocumentLength int64
+	url               string
+	maxRedirect       int
+}
+
+type ScrapeBuilder interface {
+	SetUserAgent(string) ScrapeBuilder
+	SetMaxDocumentLength(int64) ScrapeBuilder
+	SetUrl(string) ScrapeBuilder
+	SetMaxRedirect(int) ScrapeBuilder
+	GetScraper() (*Scraper, error)
+}
+
+type scrapeBuilder struct {
+	scrapeSettings scrapeSettings
+}
+
+func (b *scrapeBuilder) GetScraper() (*Scraper, error) {
```

Comment:
I don't know what is more common in `go` land - to have `GetXXX` from `XXXBuilder` or to have a generic `Build`, please check in other libraries

Reply:
In some places find that is more used `Build`.